pkg/build/api/types.go

```diff
@@ -12,6 +12,8 @@ const (
 	BuildAnnotation = "openshift.io/build.name"
 	// BuildLabel is the key of a Pod label whose value is the Name of a Build which is run.
 	BuildLabel = "build"
+	// BuildIndicatorLabel is the key of a Pod label which can be used to query for all build pods
+	IsBuildPodLabel = "isBuildPod"
```

Comment:
You can query on anything you write code for.  Label query is no more efficient than any other type of query.  That query at the end is pretty bad.  Also... build name == pod name so why in the world would we need to list anything?

----- Original Message -----
> > @@ -12,6 +12,8 @@ const (
> >  	BuildAnnotation = "openshift.io/build.name"
> >  	// BuildLabel is the key of a Pod label whose value is the Name of a
> >  	Build which is run.
> >  	BuildLabel = "build"
> > +	// BuildIndicatorLabel is the key of a Pod label which can be used to
> > query for all build pods
> > +	IsBuildPodLabel = "isBuildPod"
> 
> can't query on annotations, right?  i do this later:
> 
> 	selector := labels.LabelSelector{}.Add(buildapi.IsBuildPodLabel,
> 	labels.EqualsOperator, []string{"true"})
> 	podList, err := lw.client.KubeClient.Pods(kapi.NamespaceAll).List(selector,
> 	fields.Everything())
> 
> 
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/2465/files#r30928406